transcom/mymove
# INT B-19905 read only mode when move is locked

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19905)

## Summary

The final piece of the puzzle to allow for full functionality of locking moves while another user is working on them is to assess all buttons an office user can use and disable them. This is a way to make the move "read only" so no changes can be made while locked by another use.

This PR does the following:
- small update to the backend to fix an oversight that checks for moves NOT in the office user's queue/GBLOC that are locked by the user when going back to their queue
- checks for `move_lock` flag at the most parent component for office users (`TXOMoveInfo.jsx`) and then passes that value down to all relevant components as a parameter. Doing it this way keeps the app from having to have excessive `useEffects` and we can just call it once when the move loads and pass it around
- adds parameter in several places to either conditionally remove buttons or disable them if `isMoveLocked` is `true`
- updates and adds relevant tests

### How to test
**NOTE** you can use an incognito browser to simulate another office user locking a move

1. In your `envrc`, turn `FEATURE_FLAG_MOVE_LOCK` to `true`
2. Fire up your `server` and `client`
3. In an incognito browser, open up MM as an office user and open up a move
4. In your non-incognito browser log in as an office user in the same GBLOC and sign in and click into the move that has the lock icon
5. Navigate through the locked move and make sure you can't make any changes to it
6. Change roles to TIO, SC, QAE, etc and repeat steps 3-5

## Screenshots
All Move screens as TXO (download video)
https://github.com/transcom/mymove/assets/136510600/f7c8ef3e-1ae6-43cb-868c-74cff3b34f07

Disabled buttons as QAE role
![Screenshot 2024-05-17 at 8 50 35 AM](https://github.com/transcom/mymove/assets/136510600/ba350e93-05bd-4ade-b6bc-f7140d2a2f09)
